### PR TITLE
Build minorminer's C docs

### PR DESCRIPTION
**Pull request by JoelPasvolsky**
Requires minorminer release with https://github.com/dwavesystems/minorminer/commit/95fcf55233931826d47134b5bbcefd05fc01f835

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -277,3 +290,9 @@ def setup(app):
 
 # show inherited members
 # autodoc_default_flags = ['members', 'undoc-members', 'inherited-members', 'show-inheritance']
+
+read_the_docs_build = os.environ.get('READTHEDOCS', None) == 'True'
+
+if read_the_docs_build:
+
+    subprocess.call('cd ../minorminer/docs/; make cpp', shell=True)
```
Comment (randomir) [inline docs/conf.py:298]: Is there any reason not to put this in [`.readthedocs.yml`](https://docs.readthedocs.io/en/stable/config-file/v2.html)?

I would not recommend we run `make` on module import. :see_no_evil:

**Comment by JoelPasvolsky** [comment]
The `ci/circleci: test-doctest` failure is due to breathe importing an obsolete sphinx extension. I set the same requirement on breathe that is in https://github.com/dwavesystems/minorminer/blob/main/docs/requirements.txt: breathe<4.13.0. This problem is fixed in breathe versions 4.14.1 and up. Locally I run doctest successfully with version 4.23.0. @boothby, what's the reason for the upper limit on breathe in minorminer? Can I up it a tad?

**Comment by boothby** [comment]
Great question... git blames you for that :stuck_out_tongue_closed_eyes:

**Comment by JoelPasvolsky** [comment]
Well that's embarrassing. I see that at the time we had an older version of sphinx in minorminer so probably we can update it there too. Here sphinx is not pinned at all so I'm unpinning breathe too. Thanks @boothby!

File: docs/conf.py (hunk 1 above)
Comment (randomir) [reply docs/conf.py:298]: Ups. On a closer look, RtD doesn't support custom (pre-)build steps. They do have a [feature request opened](https://github.com/readthedocs/readthedocs.org/issues/6662), likely to be implemented at some point.

I expected it's possible to customize the [requirements install](https://docs.readthedocs.io/en/stable/config-file/v2.html#python) a bit more. Also, to use a custom docker image. But no. Pretty disappointing. :stuck_out_tongue: 

There is [this hackish workaround](https://github.com/readthedocs/readthedocs.org/issues/6662#issuecomment-653675848) to put a build step in a separate yaml file, but I'm not sure that's any better than running from `conf.py`.

Comment (JoelPasvolsky) [reply docs/conf.py:298]: Thanks @randomir!

**Comment by randomir** [comment]
@JoelPasvolsky, is this ready for merge?

**Comment by JoelPasvolsky** [comment]
@randomir, yup, go for it.

**Pull request merged**